### PR TITLE
docs: Terraform RBAC Tutorial

### PR DESCRIPTION
**Pull request by dehume**
### Motivation

Including a tutorial (based on [rbac tutorial](https://materialize.com/docs/manage/access-control/rbac-tutorial/)) explaining how to do the same workflow using Terraform grant resources. This will cover three of the four grant resources:

* roles
* object grant
* role membership

It does not currently cover default grant resources as the rbac tutorial does not include that. Can include if we are fine with the tutorials differing.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: doc/user/content/manage/access-control/rbac-terraform-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+{{< private-preview />}}
+
+This tutorial walks you through managing roles in Materialize with [Terraform](https://www.terraform.io/). By the end of this tutorial you will:
+
+* Create two new roles in your Materialize
+* Apply privileges to the new roles
+* Assign a role to a user
+* Modify and remove privileges on roles
+
+In this scenario, you are an DevOps engineer responsible for managing your Materialize account with code. You recently hired a new developer who needs privileges in a non-production cluster. You will create specific privileges for the new role that align with your business needs and restrict the developer role from having access to your production cluster.
```
Comment (sjwiesman) [inline doc/user/content/manage/access-control/rbac-terraform-tutorial.md:19]:
```suggestion
In this scenario, you are a DevOps engineer responsible for managing your Materialize account with code. You recently hired a new developer who needs privileges in a non-production cluster. You will create specific privileges for the new role that align with your business needs and restrict the developer role from having access to your production cluster.
```

File: doc/user/content/manage/access-control/rbac-terraform-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+* Make sure you have a [Materialize account](https://materialize.com/register/?utm_campaign=General&utm_source=documentation) and already have a password to connect with.
+
+* You should be familiar with setting up a [Terraform project in Materialize](https://materialize.com/docs/manage/terraform/).
+
+* Have an understanding of permissions in Materialize. These grant resources in this tutorial are modeled after the [access-control tutorial](https://materialize.com/docs/manage/access-control/rbac-tutorial/).
+
+## Step 1. Create Role
+
+1. A role is a collection of privileges you can apply to users. First we will create a role resource in Terraform.
```
Comment (sjwiesman) [inline doc/user/content/manage/access-control/rbac-terraform-tutorial.md:31]:
```suggestion
1. A role is a collection of privileges you can apply to users. First, we will create a role resource in Terraform.
```

File: doc/user/content/manage/access-control/rbac-terraform-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+    SELECT name, privileges FROM mz_tables WHERE name = 'dev_table';
+    ```
+
+    ```nofmt
+    name|privileges
+    dev_table|{u1=arwd/u1,u8=arw/u1}
+    (1 row)
+    ```
+
+Currently the `dev_role` does not have any permissions on the table `dev_table`.
```
Comment (sjwiesman) [inline doc/user/content/manage/access-control/rbac-terraform-tutorial.md:128]:
```suggestion
Currently, the `dev_role` has no permissions on the table `dev_table`.
```

**Review by sjwiesman — COMMENTED** [review]
Left just a few grammar nits

File: doc/user/content/manage/access-control/rbac-terraform-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+    ```
+
+2. We will run Terraform to create this role.
+
+    ```shell
+    terraform apply
+    ```
+
+    {{< note >}}
+    All of the resources in this tutorial can be run with a single `terraform apply` but we will add and apply resources incrementally to better illustrate grants.
```
Comment (joacoc) [inline doc/user/content/manage/access-control/rbac-terraform-tutorial.md:46]: Don't know why (it is a bug) but removing one tab here formats the text correctly and not as if it was code:

```
    {{< note >}}
  All of the resources in this tutorial can be run with a single terraform apply but we will add and apply resources incrementally to better illustrate grants.
    {{</ note >}}
```

File: doc/user/content/manage/access-control/rbac-terraform-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+      role_name     = materialize_role.dev_role.name
+      privilege     = "UPDATE"
+      database_name = materialize_table.table.database_name
+      schema_name   = materialize_table.table.schema_name
+      table_name    = materialize_table.table.name
+    }
+    ```
+
+    {{< note >}}
+    All of the grant resources are a 1:1 between a specific role, object and privilege. So adding three privileges to the `dev_role` will require three Terraform resources.
```
Comment (joacoc) [inline doc/user/content/manage/access-control/rbac-terraform-tutorial.md:168]: The same as the other note.

File: doc/user/content/manage/access-control/rbac-terraform-tutorial.md [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+
+    ```nofmt
+    name|privileges
+    dev_table|{u1=arwd/u1,u8=arw/u1}
+    (1 row)
+    ```
+
+Currently the `dev_role` does not have any permissions on the table `dev_table`.
+
+### Step 3. Grant privileges on example objects
```
Comment (joacoc) [inline doc/user/content/manage/access-control/rbac-terraform-tutorial.md:130]: From Step 3. to Step 6. it has an additional `#` 
```suggestion
## Step 3. Grant privileges on example objects
```

**Review by joacoc — COMMENTED** [review]
LGTM.

**Review by joacoc — APPROVED** [review]

**Review by bobbyiliev — APPROVED** [review]

**Pull request merged**